### PR TITLE
WIP: DirectML implementation

### PR DESCRIPTION
**Pull request by Axiverse**

### CODE REVIEW

File: src/Vortice.DirectML/BufferTensorDescription.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+﻿// Copyright © Amer Koleci and Contributors.
+// Licensed under the MIT License (MIT). See LICENSE in the repository root for more information.
+
+namespace Vortice.DirectML;
```
Comment (amerkoleci) [inline src/Vortice.DirectML/BufferTensorDescription.cs:4]: Can you leave space here please? Same in other files

Comment (Axiverse) [reply src/Vortice.DirectML/BufferTensorDescription.cs:4]: Done

**Comment by Axiverse** [comment]
I've updated the alloc method to be more in line with some other examples I've found. PTAL

See: OperatorDesc, TensorDesc and specializations BufferTensorDesc, ElementWiseIdentityOperatroDesc, AdamOptimizerOperatorDesc

**Comment by amerkoleci** [comment]
Thanks!

**Pull request merged**